satpy/tests/reader_tests/test_satpy_cf_nc.py

```diff
@@ -334,3 +395,31 @@ def test_orbital_parameters(self, _cf_scene, _nc_filename):
         assert isinstance(new_attrs, dict)
         for key in orig_attrs:
             assert orig_attrs[key] == new_attrs[key]
+
+    def test_write_and_read_from_two_files(self, _nc_filename, _nc_filename_i):
+        """Save two datasets with different resolution and read the solar_zenith_angle again."""
+        _create_test_netcdf(_nc_filename, resolution=742)
+        _create_test_netcdf(_nc_filename_i, resolution=371)
+        scn_ = Scene(reader='satpy_cf_nc',
+                     filenames=[_nc_filename, _nc_filename_i])
+        scn_.load(['solar_zenith_angle'], resolution=742)
+        assert scn_['solar_zenith_angle'].attrs['resolution'] == 742
+        scn_.unload()
+        scn_.load(['solar_zenith_angle'], resolution=371)
+        assert scn_['solar_zenith_angle'].attrs['resolution'] == 371
+
+    def test_dataid_attrs_equal_non_equal(self, _cf_scene):
+        """Check that attrs return False on non equal attribute and handles missing keys."""
+        from satpy.dataset.dataid import DataID, default_id_keys_config
+        reader = SatpyCFFileHandler('filename',
+                                    {},
+                                    {'filetype': 'info'})
+        ds_id = DataID(default_id_keys_config, name='image0', resolution=999, modifiers=())
+        data = _cf_scene
+        assert reader._dataid_attrs_equal(ds_id, data['image0']) is False
+        ds_id_wavelength = DataID(default_id_keys_config, name='image0', resolution=1000, modifiers=(),
+                                  wavelength=WavelengthRange(min=0.1, central=0.63, max=1000, unit='µm'))
+        assert reader._dataid_attrs_equal(ds_id_wavelength, data['image0']) is False
+        ds_id_key_error = DataID(default_id_keys_config, name='image0', resolution=1000)
+        del data['image0'].attrs['resolution']
+        assert reader._dataid_attrs_equal(ds_id_key_error, data['image0']) is True
```

Comment:
I'm not sure this test is needed :)
This is a private function which probably indicates it is an implementation detail, and we don't want to test implementation.
What we want to test though it the functionality, as you did in the previous test, so I think we're good.

Reply:
I think I understand what you mean. But then the exception in _dataid_attrs_equal is not tested.
I guess I should expand the functionality test to also test this then.

Reply:
I think the check should be the public interface `.get_dataset()` returning `None` (which should be explicit in the code, btw) for the `False` case and a dataset on `True` case. And this test should be split into three separate ones :-)

Reply:
OK, so I followed the suggestion from Panu. Lets see if I got it better.